_posts/2019-09-06-kubernetes-and-hokusai.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: epic
+title: "Kubernetes and Hokusai"
+date: "2019-09-06"
+author: [matt-dole]
+categories: [artsy, beginners, engineering, hokusai, kubernetes, k8s]
+---
+
+When I joined Artsy Engineering a few months ago, I had roughly 0 knowledge of Kubernetes. I'd heard the term
+thrown around a few times, but had no idea how it worked or what it was good for.
+
+Kubernetes is still a bit of a mystery to me, but I'm able to do a lot of Kubernetes operations quickly and easily
+thanks to an open-source tool developed at Artsy: [Hokusai](https://github.com/artsy/hokusai).
+
+In this post, I'll give some background on Kubernetes, a brief history of Hokusai and a description of its
+functionality, and some pointers for how to get started using it.
+
+<!-- more -->
+
+# What is Kubernetes?
+
+On a high level, Kubernetes is a tool designed to _orchestrate containers at scale._
+
+Let's break that down a bit. First, some helpful vocab:
+
+**Container**: Effectively code + all necessary dependencies for an application. A
+["standardized unit of software"](https://www.docker.com/resources/what-container).
+
+**Pods**: Single instance of an application or process. Consists of one or more containers, though one container
+per pod is the most common use case.
+
+**Replicas**: Multiple instances of the same application, each running on its own pod.
+
+**Node**: A physical or virtual machine that runs a pod or pods.
+
+**Cluster**: A node or nodes managed by a "master" machine (which could be a physical computer or a virtual
```

Comment:
Not sure digging into master / node distinction is necessary here if we don't want to get into the weeds with Kubernetes internals